### PR TITLE
docs: Update ZMQ for module directory structure.

### LINKED ISSUE #2190: Update ZMQ Windows Documentation
# 📚 Docs or Wiki Bug Report

### Description

 As go-modules have become common place, the documentation for getting ZMQ up and running on Windows 10 needs to be updated to reflect the directory structure changes. 

### What's the affected URL?**
https://github.com/edgexfoundry/edgex-go/blob/master/ZMQWindows.md

### Expected vs Actual Behavior**

 N/A

## 📷Screenshot 

N/A

## 🔥 Exception or Error
<pre><code>

N/A
</code></pre>

## 🌍  Your Environment

 Windows 10 Pro x64 Build 17763

### PR DESCRIPTION
**Pull request by rsdmike**
Updated docs for getting ZMQ up and running on Windows 10 for EdgeX development.

Closes #2190 

Signed-off-by: Mike Johanson <michael.johanson@intel.com>

### CODE REVIEW

**Comment by rsdmike** [comment]
recheck

File: ZMQWindows.md [hunk 1]
```diff
@@ -10,25 +12,32 @@
...
+    3.b. Copy the `libsodium.dll` and `libzmq-v120-mt-4_x_x.dll` files from step 3 into `%GOPATH%\pkg\mod\github.com\pebbe\zmq4@v1.0.0\lib`; create the lib directory as needed. Rename `libzmq-v120-mt-4_x_x.dll` to `libzmq.dll`, so that gcc can link and find `-lzmq`. If this is not done, errors like this may occur: `C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/5.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe:cannot find -lzmq`
+
+4. In any directory you prefer, clone the source of ZMQ that matches the version/commit you downloaded from AppVeyor: `git clone https://github.com/zeromq/libzmq.git`.
 
-3. In any directory you prefer, clone the source of ZMQ that matches the version/commit you downloaded from AppVeyor: `git clone https://github.com/zeromq/libzmq.git`.
+> As of Nov 2019, ZMQ 4.3.3 does not work and results in an error: https://github.com/pebbe/zmq4/issues/152/. 4.3.2 is recommended and tested on Win 10:
```
Comment (jbonafide623) [inline ZMQWindows.md]: Perhaps this note can be moved to after Point 3 and before Point 4? As I am following the doc installing ZMQ on Windows, I found this comment to be very helpful. I was thinking that if this were under Point 3, it could help ensure the correct artifact is chosen.

File: ZMQWindows.md [hunk 2]
```diff
@@ -10,25 +12,32 @@
...
+    3.b. Copy the `libsodium.dll` and `libzmq-v120-mt-4_x_x.dll` files from step 3 into `%GOPATH%\pkg\mod\github.com\pebbe\zmq4@v1.0.0\lib`; create the lib directory as needed. Rename `libzmq-v120-mt-4_x_x.dll` to `libzmq.dll`, so that gcc can link and find `-lzmq`. If this is not done, errors like this may occur: `C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/5.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe:cannot find -lzmq`
+
+4. In any directory you prefer, clone the source of ZMQ that matches the version/commit you downloaded from AppVeyor: `git clone https://github.com/zeromq/libzmq.git`.
 
-3. In any directory you prefer, clone the source of ZMQ that matches the version/commit you downloaded from AppVeyor: `git clone https://github.com/zeromq/libzmq.git`.
+> As of Nov 2019, ZMQ 4.3.3 does not work and results in an error: https://github.com/pebbe/zmq4/issues/152/. 4.3.2 is recommended and tested on Win 10:
+https://ci.appveyor.com/project/zeromq/libzmq/builds/25882500/job/s2yhbnjrdouq5qnm/artifacts
+and
+https://github.com/zeromq/libzmq/tree/a84ffa12b2eb3569ced199660bac5ad128bff1f0.
```
Comment (jbonafide623) [inline ZMQWindows.md]: Assuming this commit hash is used when someone is following this guide, would it be worth mentioning that the user to should do the following to ensure that their HEAD is pointing to that commit hash:

`git reset --hard a84ffa12`

Not sure if that completely necessary to add :)

File: ZMQWindows.md [hunk 3]
```diff
@@ -10,25 +12,32 @@
...
+> It is recommended to set these environment variables in your System or IDE so that you do not need to set them in every terminal session. This especially required if you wish to debug services that depend on ZMQ. 
+
+7. Next, navigate to `$Env:GOPATH\src\github.com\edgexfoundry\edgex-go\cmd\core-data`. *If you didn't acquire dependencies previously, run `go get -v` to acquire all dependencies*. Run `go build -v`. If it complains about not finding zmq.h from auth.go, double check the CGO_ environment variables by running `go env`.
 
-6. There should now be a `core-data.exe` file in your directory. The last step is to ensure that a copy of the DLL's `libsodium.dll` and `libzmq-v120-mt-4_x_x.dll` (yes, this one, not with the rename we did earlier) is in the same directory as your `.exe`.
+8. There should now be a `core-data.exe` file in your directory. The last step is to ensure that a copy of the DLL's `libsodium.dll` and `libzmq-v120-mt-4_x_x.dll` (yes, this one, not with the rename we did earlier) is in the same directory as your `.exe`.
```
Comment (jbonafide623) [inline ZMQWindows.md]: This is more of something I've run into while following this setup guide. After completing Step 7, I am not seeing the libsodium.dll and libzmq-v120-mt4_x_x.dll in the same directory as my core-data.exe.

With that, when I execute core-data.exe, I receive the following output:
`
$ ./core-data.exe
C:/Users/Jason/go/src/github.com/edgexfoundry/edgex-go/cmd/core-data/core-data.exe: error while loading shared libraries: libzmq-v141-mt-4_3_2.dll: cannot open shared object file: No such file or directory
`
I would assume that you are not seeing this. Any possible pointers as to why this may be?

Comment (jbonafide623) [reply ZMQWindows.md]: Just wanted to add that, when I manually copy `libsodium.dll` and `libzmq-v141-mt-4_x_x.dll` to the directory in which my `exe` is, the application runs as expected. Not sure if that is a manual step or something that should happen following the steps in the guide.

**Comment by michaelestrin** [comment]
recheck

File: ZMQWindows.md (hunk 3 above)
Comment (rsdmike) [reply ZMQWindows.md]: You are correct that this is a manual step, it will not be done for you. I will reword it to make it more clear. Thanks
